resolver.go

```diff
@@ -151,22 +158,36 @@ func setCommonFlags(msg *dns.Msg) {
 	msg.RecursionAvailable = true
 }
 
+func shuffleAddr(addr []net.IP) []net.IP {
```

Comment:
My contention is that your reasoning to state "shuffling is important" is flawed. 

Im just trying to help: we went down exactly the same road, only to realise that `getaddrinfo` defeats you.